modular/blocksyncer/blocksyncer_options.go

```diff
@@ -1,478 +0,0 @@
-			} else {
-				err = module.PrepareTables()
-			}
-			if err != nil {
-				log.Errorw("failed to PrepareTables/AutoMigrate tables", "error", err)
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-// serve start BlockSyncer rpc service
-func (b *BlockSyncerModular) serve(ctx context.Context) {
-	migrateDBAny := ctx.Value(MigrateDBKey{})
-	if migrateDB, ok := migrateDBAny.(bool); ok && migrateDB {
-		err := b.initDB(true)
-		if err != nil {
-			log.Errorw("fail to init DB", "error", err)
-			return
-		}
-	}
-	// Create a queue that will collect, aggregate, and export blocks and metadata
-	exportQueue := types.NewQueue(100)
-
-	// Create workers
-	worker := parser.NewWorker(b.parserCtx, exportQueue, 0, config.Cfg.Parser.ConcurrentSync)
-	worker.SetIndexer(b.parserCtx.Indexer)
-
-	latestBlockHeight := mustGetLatestHeight(b.parserCtx)
-	Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Store(int64(latestBlockHeight))
-	go b.getLatestBlockHeight(ctx)
-
-	lastDbBlockHeight := uint64(0)
-	for {
-		epoch, err := b.parserCtx.Database.GetEpoch(context.TODO())
-		if err != nil {
-			log.Errorw("failed to get last block height from database", "error", err)
-			continue
-		}
-		lastDbBlockHeight = uint64(epoch.BlockHeight)
-		break
-	}
-
-	// fetch block data
-	go b.quickFetchBlockData(lastDbBlockHeight + 1)
-
-	go b.enqueueNewBlocks(ctx, exportQueue, lastDbBlockHeight+1)
-
-	// Start each blocking worker in a go-routine where the worker consumes jobs
-	go worker.Start(ctx)
-}
-
-// enqueueNewBlocks enqueues new block heights onto the provided queue.
-func (b *BlockSyncerModular) enqueueNewBlocks(context context.Context, exportQueue types.HeightQueue, currHeight uint64) {
-	// Enqueue upcoming heights
-	for {
-		select {
-		case <-context.Done():
-			log.Infof("Receive cancel signal, enqueueNewBlocks routine will stop")
-			// close channel
-			close(exportQueue)
-			return
-		default:
-			{
-				latestBlockHeightAny := Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Load()
-				latestBlockHeight := latestBlockHeightAny.(int64)
-				// Enqueue all heights from the current height up to the latest height
-				for ; currHeight <= uint64(latestBlockHeight); currHeight++ {
-					// log.Debugw("enqueueing new block", "height", currHeight)
-					exportQueue <- currHeight
-				}
-			}
-		}
-	}
-}
-
-func (b *BlockSyncerModular) getLatestBlockHeight(ctx context.Context) {
-	for {
-		select {
-		case <-ctx.Done():
-			log.Infof("Receive cancel signal, getLatestBlockHeight routine will stop")
-			return
-		default:
-			{
-				latestBlockHeight, err := b.parserCtx.Node.LatestHeight()
-				if err != nil {
-					log.Errorw("failed to get last block from RPCConfig client",
-						"error", err,
-						"retry interval", config.GetAvgBlockTime())
-					continue
-				}
-				Cast(b.parserCtx.Indexer).GetLatestBlockHeight().Store(latestBlockHeight)
-
-				time.Sleep(time.Second)
-			}
-		}
-	}
-}
-
-func (b *BlockSyncerModular) quickFetchBlockData(startHeight uint64) {
```

Comment:
`quickFetchBlockData` --> `fetchBlockDataLoop`

Reply:
I'll change my name and keep the code the same